libbeat/processors/add_locale/add_locale.go

```diff
@@ -0,0 +1,47 @@
+package actions
+
+import (
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/processors"
+	"github.com/pkg/errors"
+)
+
+type addLocale struct {
+	timezone string
+}
+
+func init() {
+	processors.RegisterPlugin("add_locale", newAddLocale)
+}
+
+func newAddLocale(c common.Config) (processors.Processor, error) {
+	config := struct {
+		TimeZone string `config:"timezone"`
+	}{}
+
+	err := c.Unpack(&config)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unpack add_locale config")
+	}
+
+	l := addLocale{timezone: config.TimeZone}
+
+	return l, nil
+}
+
+func (l addLocale) Run(event common.MapStr) (common.MapStr, error) {
+	zone, err := time.LoadLocation(l.timezone)
```

Comment:
The `Run` method should be as simple and fast as possible. In `newAddLocale()` you I think can do this work once and store the result so that Run just becomes a `map.Put()` operation.